homeassistant/components/lg_netcast/config_flow.py

```diff
@@ -0,0 +1,160 @@
+                translation_key="deprecated_yaml_import_issue_already_configured",
+                translation_placeholders={
+                    "domain": DOMAIN,
+                    "interation_title": "LG Netcast",
+                },
+            )
+            raise err
+
+        async_create_issue(
+            self.hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.6.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "LG Netcast",
+            },
+        )
+
+        return await self.async_step_authorize(config)
+
+    async def async_step_authorize(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle Authorize step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None and user_input.get(CONF_ACCESS_TOKEN) is not None:
+            self.device_config[CONF_ACCESS_TOKEN] = user_input[CONF_ACCESS_TOKEN]
+
+        self.create_client()
+        assert self.client is not None
+
+        if not self.device_config.get(CONF_ID):
+            try:
+                details = await async_discover_netcast_details(self.hass, self.client)
+            except LGNetCastDetailDiscoveryError:
+                return self.async_abort(reason="cannot_connect")
+            unique_id = details["uuid"]
+            if unique_id is None:
+                return self.async_abort(reason="invalid_host")
+            self.device_config[CONF_ID] = unique_id
+            if CONF_NAME not in self.device_config:
+                model_name = details["model_name"]
+                friendly_name = details["friendly_name"] or DEFAULT_NAME
+                self.device_config[CONF_NAME] = (
+                    f"{friendly_name} ({model_name})" if model_name else friendly_name
+                )
+
+        await self.async_set_unique_id(self.device_config[CONF_ID])
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self.device_config[CONF_HOST]}
+        )
+
+        try:
+            await self.hass.async_add_executor_job(
+                self.client._get_session_id  # pylint: disable=protected-access
+            )
+            return await self.async_create_device()
+        except AccessTokenError:
+            if user_input is not None:
+                errors[CONF_ACCESS_TOKEN] = "invalid_access_token"
+        except SessionIdError:
+            errors["base"] = "cannot_connect"
+
+        return self.async_show_form(
+            step_id="authorize",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_ACCESS_TOKEN): vol.All(str, vol.Length(max=6)),
+                }
+            ),
+            errors=errors,
+        )
```

Comment:
When I re-added the device, after entering the IP address, the integration asks to enter an access token, but the TV does not show it. To see it I had to enter `0000`. Maybe we should do this automatically like the Bravia integration does (`self.client.pair` trying to connect with `0000` PIN which causes a popup on the TV)?

https://github.com/home-assistant/core/blob/e0490a3ade04fdc3fcbcacb16ce56ad308408f31/homeassistant/components/braviatv/config_flow.py#L163-L166

Reply:
When no access token is provided it is sending an auth token request already, however I have seen similar issues to what you've described. I think it's related to some timers in the TV as it is seemingly random. The way I had intended to handle this was by submitting an empty access token to force the auth to happen again, which in my testing usually works. But I note that this wasn't obvious this was possible.

I've now implemented the ability for the flow to regularly send the auth request every second whilst we're waiting for the access token to improve the user experience, which should handle both the prompt not appearing (or briefly appearing and then immediately disappearing), but also removes the time sensitivity that was there previously.

If you find that you still need to send an invalid token I can modify this to do so.